2 - Templates/extra_scripts/shape_convert.R

```
@@ -28,77 +6,164 @@ pacman::p_load(
+      # Calculate the proportion of the old area intersects with the new area.
+      prop_old_in_new = intersect_area / old_area,
+    ) %>%
+    group_by(area_code_old) %>%
     mutate(
-      prop_of_prop_inter_new = prop_inter/sum(prop_inter, na.rm = TRUE),
-      check_100_new = sum(prop_of_prop_inter_new),
-      measure_weighted = case_when(
-        type == "mean" ~ measure*prop_of_prop_inter_new,
-        type == "sum" ~ measure*prop_inter
+      # Calculate a scalar for the measure based the weight choice.
+      conversion_scalar_part = case_when(
+        weight == "none" ~ 1,
+        weight == "weighting" ~ weighting_as_prop,
+        weight == "overlap" ~ prop_old_in_new,
+        weight == "weighting and overlap" ~ weighting_as_prop * prop_old_in_new
       ),
-      measure_new = sum(measure_weighted, na.rm = TRUE),
+      
+      # - If old_shape is areas, make the scalar add to 1 for each new area.
+      # This makes it easier to calculate measure_new in the next step.
+      # - If old_shape is points, weight == "none", and we want the mean, then
+      # we treat it the same as with areas.
+      # - If old_shape is points, weight == "none", and we want the sum
+      # (aggregated into new area), then leave the conversion_scalar_part as 1
+      # so it can stack.
+      conversion_scalar = case_when(
+        weight != "none" ~ conversion_scalar_part / sum(conversion_scalar_part),
+        weight == "none" & type == "mean" ~ conversion_scalar_part / sum(conversion_scalar_part),
+        weight == "none" & type == "sum" ~ conversion_scalar_part
+      )
+    ) %>%
+    ungroup() %>%
+    group_by(area_code_new) %>%
+    mutate(
+      # First, apply scalar to each old measure.
+      measure_new_part = measure * conversion_scalar,
+      
+      # Create new weighted measure by summing or averaging these parts.
+      # This structure feels quite confusing now, but the output data looks ok.
+      measure_new = case_when(
```

Why na.rm = TRUE?

Could this mask an error?

---

I suppose it could mask a situation where a new area is made from a sum or mean of, say, three areas. The first two areas would have been 99% of overlap, but they have NA as the measure value, so the new areas measure would come from the 1st overlap area.

Also would mask a situation where a new area is the sum of, say, three areas, but two area NA so the new area has a weirdly low sum?

I could change so this function must not take any missing data (measure or weighting, if given)?